File: src/188. The hyperexponentiation of a number/188.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	var m big.Int
	m.Exp(big.NewInt(10), big.NewInt(8), nil)
	fmt.Println(hyperExp(int64(1777), int64(1855), &m))
}

func hyperExp(a int64, b int64, m *big.Int) *big.Int {
	var result big.Int
	if b == 1 {
		result.Mod(big.NewInt(a), m)
		return &result
	} else {
		result.Exp(big.NewInt(a), hyperExp(a, b - 1, m), m)
		return &result
	}
}

File: src/188. The hyperexponentiation of a number/188.java
import java.math.BigInteger;

public class Main {

    public static void main(String[] args) {
        BigInteger m = BigInteger.valueOf(100_000_000);
        System.out.println(hyperExp(1777, 1855, m));
    }

    private static BigInteger hyperExp(long a, long b, BigInteger m) {
        if (b == 1) {
            return BigInteger.valueOf(a).mod(m);
        } else {
            return BigInteger.valueOf(a).modPow(hyperExp(a, b - 1, m), m);
        }
    }

}
